src/quick-lint-js/parse.h

```diff
@@ -3006,6 +3006,19 @@ class parser {
           break;
         }
 
+        case token_type::kw_new: {
+          const char8 *here = this->lexer_.end_of_previous_token();
+          this->error_reporter_->report(error_missing_equal_after_variable{
```

Comment:
Blocking: This change will reject the following legal code:
```js
let x
new Array()
```

We should only report an error if a semicolon cannot be inserted (i.e. if there is no newline before `new`).